enhancements/rhcos/multi-stream.md

```diff
@@ -0,0 +1,467 @@
+extensions. Probably a good idea to tag osStream images uniformly and include
+stream name in the off chance that we ever grow to more than two streams.
+Whatever magic that plumbs in those image refs to MCPs needs to get updated.
+TODO: This is part that I (Scott) don't understand very well so need help.
+
+CoreOS stream files metadata section adds a `labels` array with at least
+`node.openshift.io/ID` and `node.openshift.io/version_ID` where the values
+match those fields in os-release metadata. Whenever nodes and machines are
+derived from these streams the manager for those resources should apply all
+labels from the stream metadata.
+
+** Note, it doesn't actually look like any of the current values in /etc/os-release
+are valid for to name specifically a major OS, do we need to add any? **
+
+```
+sh-5.1# cat /etc/os-release 
+NAME="Red Hat Enterprise Linux CoreOS"
+ID="rhcos"
+ID_LIKE="rhel fedora"
+VERSION="418.94.202409201939-0"
+VERSION_ID="4.18"
+VARIANT="CoreOS"
+VARIANT_ID=coreos
+PLATFORM_ID="platform:el9"
+PRETTY_NAME="Red Hat Enterprise Linux CoreOS 418.94.202409201939-0"
+ANSI_COLOR="0;31"
+CPE_NAME="cpe:/o:redhat:enterprise_linux:9::baseos::coreos"
+HOME_URL="https://www.redhat.com/"
+DOCUMENTATION_URL="https://docs.okd.io/latest/welcome/index.html"
+BUG_REPORT_URL="https://access.redhat.com/labs/rhir/"
+REDHAT_BUGZILLA_PRODUCT="OpenShift Container Platform"
+REDHAT_BUGZILLA_PRODUCT_VERSION="4.18"
+REDHAT_SUPPORT_PRODUCT="OpenShift Container Platform"
+REDHAT_SUPPORT_PRODUCT_VERSION="4.18"
+OPENSHIFT_VERSION="4.18"
+RHEL_VERSION=9.4
+OSTREE_VERSION="418.94.202409201939-0"
+```
+
+
+CoreOS stream files metadata section adds an 'eol' boolean field. Whenever the
+MCO encounters a MachineConfigPool which is configured with an osStream which
+is marked as `eol` the MCO should mark itself as Upgradeable=False with an
+appropriate message inhibiting upgrades to the next OCP minor version. In the
+unlikely event that plans change a subsequent z-stream update could be applied
+altering the `eol` value and MCO should then set Upgradeable=True whenever all
+currently in use streams are eol false.
+
+When the installer is built for OCP valid `osStream` must start with "rhcos"
+and match the name of a file in data/data/coreos/
+
+Somewhere in MCO's templates/ add streams/{rhcos-9,rhcos-10} anything outside
+of streams/ goes to MCPs of all streams. Any time there's a file in both
+non-streams and streams the more specific streams version takes precidence. And
+any file only in streams/ goes only to MCPs of the relevant stream. Whenever
+possible, prefer to implement limited conditional logic outside of streams/.
+
+Alternatively, all files deployed via MachineConfig should have internal
+conditional logic which applies to the relevant ID and VERSION_ID fields
+recorded in /etc/os-release or other available facilities. There's probably
+some systemd magic here or something. This probably also pushes more static
```

Comment:
Yes, we I think we need to make it easy dispatch on the OS major/minor; this gets into a lot of details around our use of `VERSION_ID` and how that looks...and the fact that today our versions include the OCP version and the OS version...